### PR TITLE
feat: New Service: Start Climate Control

### PR DESCRIPTION
**Pull request by coreywillwhat**
### `audiconnect.start_climate_control`

Optional inputs for:

- Target Temperature (Fahrenheit)
- Target Temperature (Celsius)
- Glass Surface Heating
- Auto Seat Comfort
  - Front Left
  - Front Right
  - Rear Left
  - Rear Right

_Note; These functions directly relate to the Audi App "Climate Control" settings for Target Temperature, Glass Surface Heating, and Auto Seat Comfort._

### services.yaml

- added new service call

### readme.md

- small improvements
- added new service call info

### Translations

- Added new service call info

### Notes

By preserving the legacy set_climatisation function, we both avoid breaking changes in existing configurations, and keep the `audiconnect.execute_vehicle_action` service simple.

### CODE REVIEW

File: custom_components/audiconnect/audi_services.py [hunk 1]
```diff
@@ -577,6 +577,81 @@ async def set_climatisation(self, vin: str, start: bool):
...
+            target_temperature = int(((temp_f - 32) * (5 / 9)) * 10 + 2731)
+        elif temp_c is not None:
+            target_temperature = int(temp_c * 10 + 2731)
+
+        # Default Temp
+        target_temperature = target_temperature or 2941
+
+        # Construct Zone Settings
+        zone_settings = [
+            {"value": {"isEnabled": seat_fl, "position": "frontLeft"}},
```
Comment (cdnninja) [inline custom_components/audiconnect/audi_services.py:602]: With these not being mandatory or defaults set wouldn't this pass none or null to the api?

File: custom_components/audiconnect/audi_account.py [hunk 2]
```diff
@@ -194,6 +221,29 @@ async def execute_vehicle_action(self, service):
...
+    async def start_climate_control(self, service):
+        _LOGGER.info("Initiating Start Climate Control Service...")
+        vin = service.data.get(CONF_VIN).lower()
+        # Optional Parameters
+        temp_f = service.data.get(CONF_CLIMATE_TEMP_F, None)
+        temp_c = service.data.get(CONF_CLIMATE_TEMP_C, None)
+        glass_heating = service.data.get(CONF_CLIMATE_GLASS, False)
+        seat_fl = service.data.get(CONF_CLIMATE_SEAT_FL, False)
+        seat_fr = service.data.get(CONF_CLIMATE_SEAT_FR, False)
+        seat_rl = service.data.get(CONF_CLIMATE_SEAT_RL, False)
```
Comment (coreywillwhat) [inline custom_components/audiconnect/audi_account.py:233]: Defaults are defined here

File: custom_components/audiconnect/audi_services.py (hunk 1 above)
Comment (coreywillwhat) [reply custom_components/audiconnect/audi_services.py:602]: Defaults are defined in `[audi_account.py](https://github.com/audiconnect/audi_connect_ha/pull/315/files/fecd5af073b6f07eb38d4515c562a792c2354476#diff-c3f638333dff9dd695ea6ad54cb0a4f1b07b71f93c533a62de37b73fc9a0f8ab#L230-234)`

Comment (cdnninja) [reply custom_components/audiconnect/audi_services.py:602]: Correct.  Thanks for pointing me to that!

Comment (coreywillwhat) [reply custom_components/audiconnect/audi_services.py:602]: No worries! In retrospect it probably makes more sense to put it in the `audi_services.py` for the exact reason you didn't see it haha

But I'll note it as a to-do item when I have some time to test it out.

**Comment by Kolbi** [comment]
Needs some conflicts to be resolved before merging.

**Comment by coreywillwhat** [comment]
> Needs some conflicts to be resolved before merging.

All set

**Pull request merged**